pkgs/development/tools/kustomize/default.nix

```diff
@@ -1,29 +1,29 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{ lib, buildGoModule, fetchFromGitHub, tree }:
 
 buildGoModule rec {
   pname = "kustomize";
-  version = "3.1.0";
-  # rev is the 3.1.0 commit, mainly for kustomize version command output
-  rev = "95f3303493fdea243ae83b767978092396169baf";
+  version = "3.3.1";
+  # rev is the 3.3.1 commit, mainly for kustomize version command output
+  # rev = "f2ac5a2d0df13c047fb20cbc12ef1a3b41ce2dad";
 
-  goPackagePath = "sigs.k8s.io/kustomize";
-  subPackages = [ "cmd/kustomize" ];
-
-  buildFlagsArray = let t = "${goPackagePath}/v3/pkg/commands/misc"; in ''
-    -ldflags=
-      -s -X ${t}.kustomizeVersion=${version}
-         -X ${t}.gitCommit=${rev}
-         -X ${t}.buildDate=unknown
-  '';
+  # they changed how the version is injected, not sure how to bump the version
+  #buildFlagsArray = let t = "sigs.k8s.io/kustomize/v3/pkg/commands/misc"; in ''
+  #  -ldflags=
+  #    -s -X sigs.k8s.io/kustomize/kustomize/v3/internal/commands.version=${version}
+  #       -X ${t}.gitCommit=${rev}
+  #       -X ${t}.buildDate=unknown
+  #'';
```

Comment:
regarding the version I belive the previous build flag was overriding [this value](https://github.com/kubernetes-sigs/kustomize/blob/v3.1.0/pkg/commands/misc/version.go#L28). This seems to be [calling different package now](https://github.com/kubernetes-sigs/kustomize/blob/v3.3.1/kustomize/internal/commands/version/version.go#L22) and there its seems to be [renamed to just `version`](https://github.com/kubernetes-sigs/kustomize/blob/v3.3.1/provenance/provenance.go#L13).

So I tried `sigs.k8s.io/kustomize/v3/provenance` and vuala - it seems to work now.

```suggestion
  buildFlagsArray = let t = "sigs.k8s.io/kustomize/v3/provenance"; in ''
    -ldflags=
      -s -X ${t}.version=${version}
         -X ${t}.gitCommit=${rev}
         -X ${t}.buildDate=unknown
  '';
```

Reply:
![image](https://user-images.githubusercontent.com/2130305/67165344-111f3000-f384-11e9-9d65-e3b715427c34.png)

Reply:
still wrong though:
```
[12:55:59] jon@jon-desktop /home/jon/projects/nixpkgs (bump-kustomize)
$ ./result/bin/kustomize version
Version: {Version:unknown GitCommit:$Format:%H$ BuildDate:1970-01-01T00:00:00Z GoOs:linux GoArch:amd64}
```

Reply:
.... that's really weird

Reply:
:see_no_evil: it's because I fu*ked it up, sorry https://github.com/NixOS/nixpkgs/pull/71453/files#r336794630